### PR TITLE
Migrate `legacy` custom properties from `v10` to `v11`

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Part of #8405 

Running this migration again after the revert of #8426. 

**Changes from last PR**
- Replacement values are in `px` instead of `rem`
- Corrects replacement value of `--p-control-border-width`
- Includes 3 manual migrations

### WHAT is this pull request doing?

Runs the [`v11-styles-replace-custom-property-legacy`](https://github.com/Shopify/polaris/pull/8272) from `@shopify/polaris-migrator` to migrate `legacy` custom properties from `v10` to `v11`. e.g.

```sh
cd polaris-react
npx @shopify/polaris-migrator@0.0.0-snapshot-release-20230224172519 v11-styles-replace-custom-property-legacy './src/**/*.scss'
npx prettier --write "./src/**/*.scss"
yarn lint:styles
```

### CODE REVIEW

File: polaris-react/src/components/Columns/Columns.scss [hunk 1]
```diff
@@ -5,10 +5,18 @@
...
+  );
+  --pc-columns-grid-template-columns-md: var(
+    --pc-columns-grid-template-columns-sm
+  );
+  --pc-columns-grid-template-columns-lg: var(
+    --pc-columns-grid-template-columns-md
+  );
+  --pc-columns-grid-template-columns-xl: var(
+    --pc-columns-grid-template-columns-lg
+  );
```
Comment (lgriffee) [inline polaris-react/src/components/Columns/Columns.scss:19]: Just formatting caught with `npx prettier --write "./src/**/*.scss"`

File: polaris-react/src/components/Checkbox/Checkbox.scss [hunk 2]
```diff
@@ -67,7 +66,7 @@
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
   height: 100%;
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-  @include focus-ring($border-width: var(--p-control-border-width));
+  @include focus-ring($border-width: var(--p-border-width-2));
```
Comment (lgriffee) [inline polaris-react/src/components/Checkbox/Checkbox.scss:69]: Manual migration

File: polaris-react/src/components/OptionList/components/Checkbox/Checkbox.scss [hunk 3]
```diff
@@ -67,7 +66,7 @@
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
   height: 100%;
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-  @include focus-ring($border-width: var(--p-control-border-width));
+  @include focus-ring($border-width: var(--p-border-width-2));
```
Comment (lgriffee) [inline polaris-react/src/components/OptionList/components/Checkbox/Checkbox.scss:69]: Manual migration

File: polaris-react/src/components/RadioButton/RadioButton.scss [hunk 4]
```diff
@@ -98,7 +96,7 @@
   }
 
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-  @include focus-ring($border-width: var(--p-control-border-width));
+  @include focus-ring($border-width: var(--p-border-width-2));
```
Comment (lgriffee) [inline polaris-react/src/components/RadioButton/RadioButton.scss:99]: Manual migration

**Review by aaronccasanova — APPROVED** [review]

**Review by alex-page — APPROVED** [review]

**Pull request merged**